client/src/router/authUtils.js

```diff
@@ -1,16 +1,49 @@
 import axios from "axios";
 import config from "../../config/config.json";
-import Cookies from "js-cookie";
+import { jwtDecode } from "jwt-decode";
 
-export const checkCookie = async () => {
-  const token = Cookies.get("token");
+const isCookieValid = (cookie, signInTime) => {
+  const token = jwtDecode(cookie);
 
-  if (token != null || token != undefined) {
-    const res = await axios.get(`${config.backend_url}/api/verifyUser`, {
-      withCredentials: true,
-    });
+  if (token.user_type && token.user_type == "regular") {
+    //check hashed_email field exists in token, if
+    //user_id field is Number type and if iat is equal
+    //to sign in time set by auth page
+    if (
+      token.hashed_email &&
+      typeof token.user_id === Number &&
+      token.iat == signInTime
+    )
+      return true;
+  } else if (token.user_type && token.user_type == "guest") {
+    //check if user_id field starts with sess
+    //and if iat is equal to sign in time
+    if (
+      token.user_id &&
+      token.user_id.startsWith("sess") &&
+      token.iat == signInTime
+    )
+      return true;
+  }
+  return false;
+};
 
-    if (res.status === 200) return true;
+export const checkCookie = async (cookie, signInTime) => {
+  if (isCookieValid(cookie, signInTime)) {
```

Comment:
you can check `authStore` existence inside this if statement. if `authStore` exists, check its time, otherwise call `verifyUser`

Reply:
I can't access authStore in this plain js file, cos it's loaded before Pinia/authStore, so I'm passing signInTime(authStore) as a parameter. In router, I'm checking if authStore doesn't exist, if it doesn't it kicks back to homepage and checkCookie won't be ran.